pkg/lib/catalogsource/image_template.go

```diff
@@ -171,14 +171,15 @@ func UpdateKubeVersion(serverVersion *version.Info, logger *logrus.Logger) {
 		return
 	}
 
-	templateNameToReplacementValuesMap[TemplKubeMajorV] = serverVersion.Major
-	templateNameToReplacementValuesMap[TemplKubeMinorV] = serverVersion.Minor
-
-	// api server does not explicitly give patch value, so we have to resort to parsing the git version
-	serverGitVersion, err := semver.Parse(serverVersion.GitVersion)
+	// need to use the gitversion from version.info.String() because minor version is not always Uint value
+	// and patch version is not returned as a first class field
+	semver, err := semver.Parse(strings.Split(strings.TrimPrefix(serverVersion.String(), "v"), "-")[0])
```

Comment:
Just a quick thought - the apimachinery util package has [parsing functions available](https://pkg.go.dev/k8s.io/apimachinery@v0.21.3/pkg/util/version#ParseSemantic) that output a struct, and methods on that struct for retrieving major/minor/patch fields as uints, so you wouldn't need to also convert the string representation to a uint later.